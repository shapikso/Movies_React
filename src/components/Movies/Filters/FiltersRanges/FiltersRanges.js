import React from 'react';
import PropTypes from 'prop-types';
import DoubleRange from '../DoubleRange/DoubleRange';
import {StFilterRow} from "../styled";

const FiltersRanges = ({ budgetMax, budgetMin, popularityMax, popularityMin, revenueMax, revenueMin,setFilter}) => {

    const setBudget = (minBudget, maxBudget) => {
        setFilter('budget_min', minBudget);
        setFilter('budget_max', maxBudget);

    };

    const setPopularity = (minPopularity, maxPopularity) => {
        setFilter('popularity_min', minPopularity);
        setFilter('popularity_max', maxPopularity);
    };

    const setVote = (minVote, maxVote) => {
        setFilter('revenue_min', minVote);
        setFilter('revenue_max', maxVote);
    };

    return (
        <StFilterRow>
            <DoubleRange setRange={setBudget}
                title="Budget, $"
                step="100000"
                minValue="0"
                maxValue="500000000"
                maxInputValue={budgetMax}
                minInputValue={budgetMin}
            />
            <DoubleRange setRange={setPopularity}
                title="Popularity"
                step="1" minValue="0"
                maxValue="200"
                maxInputValue={popularityMax}
                minInputValue={popularityMin}
            />
            <DoubleRange setRange={setVote}
                title="Vote count"
                step="1"
                minValue="0"
                maxValue="30000"
                maxInputValue={revenueMax}
                minInputValue={revenueMin}
            />
        </StFilterRow>
    );
};

FiltersRanges.propTypes = {
    budgetMax: PropTypes.number.isRequired,
    budgetMin: PropTypes.number.isRequired,
    popularityMax: PropTypes.number.isRequired,
    popularityMin: PropTypes.number.isRequired,
    revenueMax: PropTypes.number.isRequired,
    revenueMin: PropTypes.number.isRequired,
    setFilter : PropTypes.func.isRequired
};

export default FiltersRanges;