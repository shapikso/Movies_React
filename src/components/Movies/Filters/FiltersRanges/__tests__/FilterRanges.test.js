import React from 'react';
import {shallow, mount} from 'enzyme';
import FiltersRanges from "../FiltersRanges";

describe('FiltersRanges', ()=> {
    const props = {
        setFilter: jest.fn(),
        budgetMax: 50000,
        budgetMin: 0,
        popularityMax:2000,
        popularityMin: 0,
        revenueMax: 200,
        revenueMin: 0,
    };
    it('should render correctly', () => {
        const component = shallow(<FiltersRanges {...props} />);
        expect(component).toMatchSnapshot();
    });
    it('should checked render max budget double range', () => {
        const component = mount(<FiltersRanges {...props} />);
        expect(component.find('DoubleRange').at(0).getElement().props.maxInputValue).toEqual(props.budgetMax);
    });
    it('should checked render min budget double range', () => {
        const component = mount(<FiltersRanges {...props} />);
        expect(component.find('DoubleRange').at(0).getElement().props.minInputValue).toEqual(props.budgetMin);
    });
    it('should checked render max popularity double range', () => {
        const component = mount(<FiltersRanges {...props} />);
        expect(component.find('DoubleRange').at(1).getElement().props.maxInputValue).toEqual(props.popularityMax);
    });
    it('should checked render min popularity double range', () => {
        const component = mount(<FiltersRanges {...props} />);
        expect(component.find('DoubleRange').at(1).getElement().props.minInputValue).toEqual(props.popularityMin);
    });
    it('should checked render max vote count double range', () => {
        const component = mount(<FiltersRanges {...props} />);
        expect(component.find('DoubleRange').at(2).getElement().props.maxInputValue).toEqual(props.revenueMax);
    });
    it('should checked render min vote count double range', () => {
        const component = mount(<FiltersRanges {...props} />);
        expect(component.find('DoubleRange').at(2).getElement().props.minInputValue).toEqual(props.revenueMin);
    });
    it('should changed Budget ', () => {
        const component = mount(<FiltersRanges {...props} />);
        component.find('DoubleRange').at(0).getElement().props.setRange(0,30000);
        expect(props.setFilter).toHaveBeenCalledWith('budget_min',0);
        expect(props.setFilter).toHaveBeenCalledWith('budget_max',30000);
    });
    it('should changed Popularity ', () => {
        const component = mount(<FiltersRanges {...props} />);
        component.find('DoubleRange').at(1).getElement().props.setRange(0,200);
        expect(props.setFilter).toHaveBeenCalledWith('popularity_min',0);
        expect(props.setFilter).toHaveBeenCalledWith('popularity_max',200);
    });
    it('should changed Vote ', () => {
        const component = mount(<FiltersRanges {...props} />);
        component.find('DoubleRange').at(2).getElement().props.setRange(0,300);
        expect(props.setFilter).toHaveBeenCalledWith('revenue_min',0);
        expect(props.setFilter).toHaveBeenCalledWith('revenue_max',300);
    });
});