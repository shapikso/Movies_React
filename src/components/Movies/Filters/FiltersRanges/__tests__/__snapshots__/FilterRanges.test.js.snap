// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersRanges should render correctly 1`] = `ShallowWrapper {}`;
